content/reference/sql/server/debugging.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Debugging
+---
+
+# Debugging
+
+Debugging a running Dolt server can be challenging. This document covers the debugging basics and how to diagnose what is happening from common symptoms.
+
+# Basics
+
+## Make sure you are running the latest Dolt version
+
+Dolt is constantly evolving. We release a new Dolt approximately once a week. Connect to the server and run `dolt version`. Make sure the version matches the latest as seen on the [GitHub releases page](https://github.com/dolthub/dolt/releases). 
+
+To upgrade the server, download the latest Dolt binary for your platform, replace the current Dolt binary with the current one, and restart the Dolt server using `dolt sql-server`.
+
+## Examine your CPU, Memory, and Disk usage
+
+Dolt consumes CPU, Memory, and Disk. Consuming more of any of these resources than the host has available can lead to degraded performance. Use your system's built in resource monitoring systems to inspect Dolt's usage of these resources. You may need a larger host or additional [read replicas](./replication.md) to support your load.
+
+## Set your log level to TRACE
+
+To see queries being run against the server, query results, and query latency set your Dolt log level to `trace`. This can be done by starting the server like so `dolt sql-server --loglevel=trace` or by setting `log_level: trace` in your `config.yaml`.
+
+## EXPLAIN for complex queries
+
+Dolt supports the SQL `EXPLAIN` operation in order for you to see the plan for complex queries. Rearranging your query to perform fewer `JOIN`s or make better use of indexes can help speed up complex queries.
```

Comment:
An example would be good here to show what to look for.

An indexed join will have `IndexedTableAccess` for example